### PR TITLE
Update to React 17.0.1

### LINKED ISSUE #149: Update to React 17.0.1
# Chore

## Describe the chore

Including the peer dependency. Right now it restricts use with React 17

### PR DESCRIPTION
**Pull request by OrhanTozan**
## What kind of change does this PR introduce?

Use latest React version and lift React v17 restriction

This closes #149.

### CODE REVIEW

**Comment by OrhanTozan** [comment]
Not sure by the way if package-lock.json needs to be updated as well.

File: package.json [hunk 1]
```diff
@@ -17,12 +17,12 @@
     "build:icons": "node --experimental-modules internals/populate-icons.js"
   },
   "peerDependencies": {
-    "react": "^16.13.1",
-    "react-dom": "^16.13.1"
+    "react": "^17.0.1",
```
Comment (MildTomato) [inline package.json:20]: I think here we need to add in 16 and 17 as a lot of our apps in supabase are still on react 16.

something like..
```
"peerDependencies": {
    "react": "^16.13.1 || ^17.0.1",
    "react-dom": "^16.13.1 || ^17.0.1"
  },
  ```
 
 i'll need to test this locally as well, make sure it doesn't break anything.

**Review by MildTomato — CHANGES_REQUESTED** [review]
> Not sure by the way if package-lock.json needs to be updated as well.

yea you can commit the package-lock.json as well 😄

File: package.json (hunk 1 above)
Comment (OrhanTozan) [reply package.json:20]: Sounds good, will update it.

**Comment by OrhanTozan** [comment]
@MildTomato I'm not sure which values need to be updated and which shouldn't in the package-lock.json (there are some dependencies that also depend on the react version).

**Comment by MildTomato** [comment]
> @MildTomato I'm not sure which values need to be updated and which shouldn't in the package-lock.json (there are some dependencies that also depend on the react version).

@OrhanTozan sure, don't worry about it. 
I'll merge this in later and test it a few times before releasing it.

**Review by MildTomato — APPROVED** [review]

**Comment by OrhanTozan** [comment]
@MildTomato Any luck?

**Comment by MildTomato** [comment]
sorry @OrhanTozan ! 

testing this locally now :D

**Comment by MildTomato** [comment]
@OrhanTozan released in v0.21.0 

thanks for the PR 👍

**Comment by OrhanTozan** [comment]
😄

**Pull request merged**